File: Angular/src/app/shared/models/institution/CourseResponse.ts
import {CoursePostResponse} from './CoursePostResponse';
import {Quiz} from '../Quiz';

export interface CourseResponse {
    id?: string;
    name?: string;
    description?: string;
    credit?: number;
    teacher?: string;
    group?: string;
    institutionID?: string;
    posts?: CoursePostResponse[];
    quizzes?: Quiz[];
}

File: Spring Boot/Courzelo/src/main/java/org/example/courzelo/dto/responses/CourseResponse.java
package org.example.courzelo.dto.responses;

import lombok.Builder;
import lombok.Data;
import org.example.courzelo.dto.QuizDTO;

import java.util.List;

@Data
@Builder
public class CourseResponse {
    private String id;
    private String name;
    private String description;
    private int credit;
    private String teacher;
    private List<String> students;
    private String group;
    private String institutionID;
    private List<CoursePostResponse> posts;
    private List<QuizDTO> quizzes;
}
